shared/actions/notifications.js

```diff
@@ -2,45 +2,85 @@
 import * as Constants from '../constants/notifications'
 import ListenerCreator from '../native/notification-listeners'
 import engine from '../engine'
-import setNotifications from '../util/set-notifications'
-import type {Dispatch} from '../constants/types/flux'
-import type {LogAction, NotificationKeys, NotificationAction} from '../constants/notifications'
-import type {Text as KBText, LogLevel} from '../constants/types/flow-types'
 import {NotifyPopup} from '../native/notifications'
 import {log} from '../native/log/logui'
+import {notifyCtlSetNotificationsRpc} from '../constants/types/flow-types'
+import {registerIdentifyUi, setupUserChangedHandler} from './tracker'
+import {call, put, take} from 'redux-saga/effects'
+
+import type {LogAction, NotificationKeys, ListenForNotifications, BadgeAppAction} from '../constants/notifications'
+import type {SagaGenerator} from '../constants/types/saga'
+import type {Text as KBText, LogLevel} from '../constants/types/flow-types'
 
-export function logUiLog ({text, level}: {text: KBText, level: LogLevel}, response: any): LogAction {
+function logUiLog ({text, level}: {text: KBText, level: LogLevel}, response: any): LogAction {
   log({text, level}, response)
   return {type: Constants.log, payload: {text: text.data, level}}
 }
 
-var initialized = false
-export function listenForNotifications (): (dispatch: Dispatch) => void {
-  return (dispatch, getState) => {
-    if (initialized) {
-      return
-    }
-
-    setNotifications({
-      session: true,
-      users: true,
-      kbfs: true,
-      service: true,
-      app: true,
-      pgp: true,
+function listenForNotifications (): ListenForNotifications {
+  return {type: Constants.listenForNotifications, payload: undefined}
+}
+
+function * _listenSaga (): SagaGenerator<any, any> {
+  const channels = {
+    app: true,
+    chat: false,
+    favorites: false,
+    kbfs: true,
+    kbfsrequest: false,
+    keyfamily: false,
+    paperkeys: false,
+    pgp: true,
+    service: true,
+    session: true,
+    tracking: true,
+    users: true,
+  }
+
+  engine().listenOnConnect('setNotifications', () => {
```

Comment:
🐑 

This could be using `call` instead, so it doesn't do any side effects within the saga.

something like

```
const engineInst = yield call(engine)
yield call([engineInst, engineInst.listenOnConnect], 'setNotifications', () => {...})
```